engine/docker/container.go

```diff
@@ -57,11 +56,10 @@ func (e *Engine) VirtualizationCreate(ctx context.Context, opts *enginetypes.Vir
 
 	restartPolicy := ""
 	restartRetry := 0
-	if opts.Restart == -1 || opts.Restart > 0 {
-		restartPolicy = restartAlways
-	}
-	if opts.Restart > 0 {
-		restartRetry = opts.Restart
+	restartStr := strings.Split(opts.Restart, ":")
+	restartPolicy = restartStr[0]
+	if r, err := strconv.Atoi(restartStr[len(restartStr)-1]); err != nil {
+		restartRetry = r
```

Comment:
Why you set `retry` with `r` when the conversion failed?
And just picking the last element to convert anyway is a little tricky. Normally we just use len() == 2 or somehow.
Further more is there any to_lower() before this piece of code?

Last better to add unit test for these kind of small func.

Reply:
Also trim_space() like to_lower()